enhancements/network/netflow.md

```diff
@@ -0,0 +1,110 @@
+* Customers need to be able to see a log of the network traffic that flows thru OpenShift cluster.
+* Customers need to get metrics and monitoring data of the OpenShift cluster network traffic to make decisions on capacity and/or security issues.
+
+### Goals
+
+- Add support for exporting NetFlow data from OpenShift.
+- Add an optional collector that consumes the netflow data and enriches it with OpenShift metadata. 
+- Add Prometheus metrics from NetFlow data collected for visualization and alerting purposes.
+
+### Non-Goals
+
+- This document only applies to OVN-Kubernetes, not OpenShift SDN.
+- This document will not the flow logs store solutions. Typically, customers store the NetFlow data in a data warehouse system (e.g. ClickHouse) or search and analytics system (e.g. ElasticSearch). It is out of the scope of this document to discuss which option should be used and if we should manage that solution. 
+
+## Proposal
+
+The Cluster Network Operator (CNO) should expose options to enable the NetFlow data from OVS bridges.
+OVS supports NetFlow v5, IPFIX and sflow. NetFlow v5 does not support IPv6, therefore we should only consider as protocol options
+both IPFIX/sflow or pick one of them. Sflow is a more performant as it is a sampling protocol and does not rely on templates.
+The bridge where the containers are connected on OVN-Kubernetes is br-int and its management port is ovn-k8s-mp0, thus to enable sflow
+the command to enable sflow would be similar to:
+
+`ovs-vsctl -- --id=@sflow create sflow agent=ovn-k8s-mp0 target="\"${nf_svc_ip}:6343\"" header=128 sampling=64 polling=10 -- set bridge br-int sflow=@sflow`
+
+In the above example **nf_svc_ip** is a variable that should be configurable. If the customer has its own collector, they should be able to define it. Therefore this value
+should also be exposed as a CNO variable.
+If not provided, we deploy a collector, expose it with a service and we set the service ClusterIP as the value of **nf_svc_ip**.
+There are many open-source NetFlow collectors. Just to name a few:
+
+* Goflow
+* Vflow
+* Pmacct
+* Skydive sflow probe
+
+TODO Write down pros and cons of each one of the options
+
+In order to provide better context of NetFlow data, we should enrich the flows collected with OpenShift metadata.
+The flows contain fields for source IP address and destination IP address. We would query the Kubernetes API for those IPs and enrich the flows when applicable with:
+
+* OpenShiftSrcPod
+* OpenShiftDstPod
+* OpenShiftSrcPodNamespace
+* OpenShiftDstPodNamespace
+* OpenShiftSrcNode (when the source address is a node IP)
+* OpenShiftDstNode (when the destination address is a node IP)
+
+This enrichment could be done by either modifying the collector source code or by using an adapter container pattern, where the collector dumps the stdout of the flows
+collected to a volume and that is consumed and enriched by a Go Kubernetes client that dumps the enriched flow to stdout/file.
```

Comment:
What are some other examples of this "adapter container pattern" ?

Reply:
yeah, I can probably reference, the k8s official docs even have a page explaining the different container patterns and adapter is also explained into detail.

Reply:
As discussed, removing the optional collector and enrichment to scope just the export of flows.